src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
+            if (arguments.isEmpty()) {
+                throw noArgumentValueWasCaptured();
+            }
 
+            return (T) arguments.get(arguments.size() - 1);
+        }
     }
 
     public List<T> getAllValues() {
-        return (List) arguments;
+        return Arrays.asList((T[]) arguments.toArray());
```

Comment:
The user may alter the returned list but in fact it is not intended by the API. Therefore it should be forebidden e.g. by an returning a copy or an readonly view.